pkg/square/square_test.go

```diff
@@ -254,3 +328,71 @@ func generateBlobTxsWithNamespaces(t *testing.T, namespaces []ns.Namespace, blob
 		blobfactory.NestedBlobs(t, namespaces, blobSizes),
 	)
 }
+
+func TestSquareBlobShareRange(t *testing.T) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	txs := blobfactory.RandBlobTxsRandomlySized(encCfg.TxConfig.TxEncoder(), 10, 1000, 10).ToSliceOfBytes()
+
+	builder, err := square.NewBuilder(appconsts.DefaultMaxSquareSize, txs...)
+	require.NoError(t, err)
+
+	dataSquare, err := builder.Export()
+	require.NoError(t, err)
+
+	for pfbIdx, tx := range txs {
+		blobTx, isBlobTx := coretypes.UnmarshalBlobTx(tx)
+		require.True(t, isBlobTx)
+		for blobIdx := range blobTx.Blobs {
+			shareRange, err := square.BlobShareRange(txs, pfbIdx, blobIdx)
+			require.NoError(t, err)
+			blobShares := dataSquare[shareRange.Start : shareRange.End+1]
+			blobSharesBytes, err := rawData(blobShares)
+			require.NoError(t, err)
+			require.True(t, bytes.Contains(blobSharesBytes, blobTx.Blobs[blobIdx].Data))
+		}
+	}
+
+	// error on out of bounds cases
+	_, err = square.BlobShareRange(txs, -1, 0)
+	require.Error(t, err)
+
+	_, err = square.BlobShareRange(txs, 0, -1)
+	require.Error(t, err)
+
+	_, err = square.BlobShareRange(txs, 10, 0)
+	require.Error(t, err)
+
+	_, err = square.BlobShareRange(txs, 0, 10)
+	require.Error(t, err)
+}
+
+func TestSquareShareCommitments(t *testing.T) {
+	txs := generateOrderedTxs(10, 10, 5)
```

Comment:
in the past, we've found edge case bugs in the share splitting process by running this same test over every square, effectively fuzzing it, so I think it would be valuable to expand on the inputs to this test either here or in the future. I'm also happy to take this on, as not all follow ups to #1690 have to be handled immediatly or in this PR